src/app/clusters/boolean-state-server/boolean-state-cluster.h

```diff
@@ -0,0 +1,47 @@
+/*
+ *    Copyright (c) 2025 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#pragma once
+
+#include <app/server-cluster/DefaultServerCluster.h>
+#include <platform/DeviceInfoProvider.h>
+
+namespace chip::app::Clusters {
+
+class BooleanStateCluster : public DefaultServerCluster
+{
+public:
+    BooleanStateCluster(EndpointId endpointId);
+
+    // Server cluster implementation
+    DataModel::ActionReturnStatus ReadAttribute(const DataModel::ReadAttributeRequest & request,
+                                                AttributeValueEncoder & encoder) override;
+    CHIP_ERROR Attributes(const ConcreteClusterPath & path, ReadOnlyBufferBuilder<DataModel::AttributeEntry> & builder) override;
+
+    CHIP_ERROR
+    SetStateValue(const BooleanState::Attributes::StateValue::TypeInfo::Type & stateValue);
+
+    CHIP_ERROR
+    GetStateValue(BooleanState::Attributes::StateValue::TypeInfo::Type & stateValue) const;
```

Comment:
Same here, maybe just return a bool? Unless this has been the pattern for other clusters but bool is a really small/primitive type, so not worth the trouble IMO

```suggestion
    bool GetStateValue();
```

Reply:
Yes, you are probably right, I am returning CHIP_ERROR instead because in the previous code (the one that used the boolean-state cluster functionality from ember, the functions Get() and Set() were returning an ember status and received the parameters by reference for Get() and const reference for Set(), and also there is some code in the examples that check that return value, but you are probably right since in the new code the functions are not supposed to fail and always return CHIP_NO_ERROR, I will take another look on this.

Reply:
Addressed.